version.go

```diff
@@ -107,20 +110,30 @@ func (v *Version) Compare(other *Version) int {
 	}
 
 	// Compare the segments
-	for i := 0; i < len(segmentsSelf); i++ {
-		lhs := segmentsSelf[i]
+	// Because a constraint could have more/less specificity than the version it's
+	// checking, we need to account for a lopsided or jagged comparison
+	var matched int
+	for i, lhs := range segmentsSelf {
+		if i > len(segmentsOther)-1 {
+			// The constraint only had X places, but the version we're checking had
+			// X+1. This means our version is greater than the constraint version we're
+			// checking , ex: 1.2.3.4 > 1.2.3
+			matched = 1
+			break
```

Comment:
There may be a bug here, if the "excess" segments are all 0s, so i'll add a case in to test / handle this before merging.